rskj-core/src/main/java/org/ethereum/db/MutableRepository.java

```diff
@@ -387,6 +387,7 @@ private byte[] getAccountData(RskAddress addr) {
     }
 
     private void internalPut(byte[] key, byte[] value) {
+        if (Arrays.equals(value, mutableTrie.get(key))) return; // writes the same value
```

Comment:
Be careful with this.

If the value of thread 1 writes value X to storage key A, and thread 2 writes value Y to storage key A, then it doesn't matter what the previous value of the cell was !
There two thread cannot be executed in parallel!
 Because if Y is different from X, the end result will depend on the order.
 What we check is program INTENTIONs not what is changed or what is not changed. This is different from adding 0 to a balance, which always preserves the balance, no matter the previous value.

Reply:
Uh I get it! Very bad impl here

So there is no way to detect if the intention is "writing the same value it was already there"?